### PR TITLE
Add ROCm ci config

### PR DESCRIPTION
**Pull request by Delaunay**

### CODE REVIEW

File: config/ci-rocm.yaml [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    args:
+      - --extra-index-url
+      - https://download.pytorch.org/whl/rocm5.2/
+    constraints:
+      - gym < 0.26.0
+
+  voir:
+    --metric: step
+    --metric-interval: 1
+    --metric-skip: 9
```
Comment (breuleux) [inline config/ci-rocm.yaml:84]: Changes in torchbenchmark require this adjustment.

```suggestion
    --stop: 20
    --metric: step
    --metric-interval: 1s
    --metric-skip: 5s
```

File: config/ci-rocm.yaml [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+  plan:
+    method: per_gpu
+
+  pip:
+    args:
+      - --extra-index-url
+      - https://download.pytorch.org/whl/rocm5.2/
+
+  voir:
+    --stop: 30
```
Comment (breuleux) [inline config/ci-rocm.yaml:48]: Because I had changed this in ci-cuda.

```suggestion
    --stop: 20
```

**Comment by Delaunay** [comment]
```
2023-03-23T22:08:12.5325828Z learning_to_paint.D0 > An error occurred
2023-03-23T22:08:12.5326509Z learning_to_paint.D0 > Traceback (most recent call last):
2023-03-23T22:08:12.5328046Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/voir/phase.py", line 191, in _step_one
2023-03-23T22:08:12.5328726Z learning_to_paint.D0 >     next_phase = gen.send(ph.value)
2023-03-23T22:08:12.5329362Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/code/torchbench/voirfile.py", line 81, in instrument_probe
2023-03-23T22:08:12.5329751Z learning_to_paint.D0 >     ov.probe(
2023-03-23T22:08:12.5330424Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/voir/overseer.py", line 43, in probe
2023-03-23T22:08:12.5331312Z learning_to_paint.D0 >     return self.require(ProbeInstrument(select(selector, skip_frames=1)))
2023-03-23T22:08:12.5332868Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/ptera/selector.py", line 845, in select
2023-03-23T22:08:12.5333516Z learning_to_paint.D0 >     rval = _resolve(sel, env, count())
2023-03-23T22:08:12.5334104Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/ptera/selector.py", line 771, in _resolve
2023-03-23T22:08:12.5334540Z learning_to_paint.D0 >     el = _resolve(selector.element, env, cnt)
2023-03-23T22:08:12.5335239Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/ptera/selector.py", line 797, in _resolve
2023-03-23T22:08:12.5335697Z learning_to_paint.D0 >     name = _eval(selector.name, env)
2023-03-23T22:08:12.5336321Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/ptera/selector.py", line 700, in _eval
2023-03-23T22:08:12.5336754Z learning_to_paint.D0 >     return x.eval(env)
2023-03-23T22:08:12.5337467Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/ptera/selector.py", line 641, in eval
2023-03-23T22:08:12.5337900Z learning_to_paint.D0 >     return dict_resolver(env)(x)
2023-03-23T22:08:12.5338500Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/ptera/selector.py", line 585, in resolve
2023-03-23T22:08:12.5338939Z learning_to_paint.D0 >     co = codefind.find_code(*hierarchy, module=module or "__main__")
2023-03-23T22:08:12.5339568Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/codefind/registry.py", line 104, in find_code
2023-03-23T22:08:12.5340050Z learning_to_paint.D0 >     filename = importlib.import_module(module).__file__
2023-03-23T22:08:12.5340870Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/importlib/__init__.py", line 127, in import_module
2023-03-23T22:08:12.5341324Z learning_to_paint.D0 >     return _bootstrap._gcd_import(name[level:], package, level)
2023-03-23T22:08:12.5341687Z learning_to_paint.D0 >   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
2023-03-23T22:08:12.5342065Z learning_to_paint.D0 >   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
2023-03-23T22:08:12.5342397Z learning_to_paint.D0 >   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
2023-03-23T22:08:12.5342883Z learning_to_paint.D0 >   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
2023-03-23T22:08:12.5343273Z learning_to_paint.D0 >   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
2023-03-23T22:08:12.5343652Z learning_to_paint.D0 >   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
2023-03-23T22:08:12.5344433Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/code/torchbench/torchbenchmark/models/LearningToPaint/__init__.py", line 1, in <module>
2023-03-23T22:08:12.5344901Z learning_to_paint.D0 >     import cv2
2023-03-23T22:08:12.5345494Z learning_to_paint.D0 >   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torchbench/lib/python3.9/site-packages/cv2/__init__.py", line 5, in <module>
2023-03-23T22:08:12.5345933Z learning_to_paint.D0 >     from .cv2 import *
2023-03-23T22:08:12.5346259Z learning_to_paint.D0 > ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

**Comment by Delaunay** [comment]
```
2023-03-23T22:10:50.2583264Z   File "/opt/actions-runner/_work/milabench/milabench/output/venv/lightning/lib/python3.9/site-packages/pl_bolts/callbacks/data_monitor.py", line 8, in <module>
2023-03-23T22:10:50.2583740Z     from pytorch_lightning.utilities.apply_func import apply_to_collection
2023-03-23T22:10:50.2584014Z resnet152.0 >! 
2023-03-23T22:10:50.2584397Z ModuleNotFoundError: No module named 'pytorch_lightning.utilities.apply_func'2023-03-23T22:10:50.2583264Z   File "/opt/actions-runner/_work/milabench/milabench/output/venv/lightning/lib/python3.9/site-packages/pl_bolts/callbacks/data_monitor.py", line 8, in <module>
2023-03-23T22:10:50.2583740Z     from pytorch_lightning.utilities.apply_func import apply_to_collection
2023-03-23T22:10:50.2584014Z resnet152.0 >! 
2023-03-23T22:10:50.2584397Z ModuleNotFoundError: No module named 'pytorch_lightning.utilities.apply_func'
```

**Comment by Delaunay** [comment]
```
2023-03-23T22:10:57.9060495Z Error Report
2023-03-23T22:10:57.9060755Z ------------
2023-03-23T22:10:57.9060848Z 
2023-03-23T22:10:57.9060931Z efficientnet_b0.D0
2023-03-23T22:10:57.9061119Z ^^^^^^^^^^^^^^^^^^
2023-03-23T22:10:57.9061345Z     No traceback found
2023-03-23T22:10:57.9061585Z efficientnet_b0.D1
2023-03-23T22:10:57.9061980Z ^^^^^^^^^^^^^^^^^^
2023-03-23T22:10:57.9062181Z     No traceback found
2023-03-23T22:10:57.9062357Z regnet_y_128gf.D0
2023-03-23T22:10:57.9062539Z ^^^^^^^^^^^^^^^^^
2023-03-23T22:10:57.9062727Z     No traceback found
2023-03-23T22:10:57.9062900Z regnet_y_128gf.D1
2023-03-23T22:10:57.9063150Z ^^^^^^^^^^^^^^^^^
2023-03-23T22:10:57.9063355Z     No traceback found
2023-03-23T22:10:57.9063553Z learning_to_paint.D0
2023-03-23T22:10:57.9063792Z ^^^^^^^^^^^^^^^^^^^^
2023-03-23T22:10:57.9064073Z     ImportError: libGL.so.1: cannot open shared object file: No such file or directory
2023-03-23T22:10:57.9064278Z 
2023-03-23T22:10:57.9064341Z learning_to_paint.D1
2023-03-23T22:10:57.9064545Z ^^^^^^^^^^^^^^^^^^^^
2023-03-23T22:10:57.9064817Z     ImportError: libGL.so.1: cannot open shared object file: No such file or directory
2023-03-23T22:10:57.9065011Z 
2023-03-23T22:10:57.9065099Z resnet152.0
2023-03-23T22:10:57.9065254Z ^^^^^^^^^^^
2023-03-23T22:10:57.9065666Z     ModuleNotFoundError: No module named 'pytorch_lightning.utilities.apply_func'
2023-03-23T22:10:57.9066049Z 
2023-03-23T22:10:57.9066184Z vit_l_32.0
2023-03-23T22:10:57.9066367Z ^^^^^^^^^^
2023-03-23T22:10:57.9067059Z     ModuleNotFoundError: No module named 'pytorch_lightning.utilities.apply_func'
2023-03-23T22:10:57.9067281Z 
2023-03-23T22:10:57.9067360Z Summary
2023-03-23T22:10:57.9067551Z -------
2023-03-23T22:10:57.9067750Z     Success: 29
2023-03-23T22:10:57.9067955Z     Failures: 8
```

**Comment by Delaunay** [comment]
```
2023-03-24T19:25:51.3616278Z Error Report
2023-03-24T19:25:51.3616699Z ------------
2023-03-24T19:25:51.3616930Z 
2023-03-24T19:25:51.3617076Z efficientnet_b0.D0
2023-03-24T19:25:51.3617397Z ^^^^^^^^^^^^^^^^^^
2023-03-24T19:25:51.3617782Z     No traceback found
2023-03-24T19:25:51.3618192Z efficientnet_b0.D1
2023-03-24T19:25:51.3618497Z ^^^^^^^^^^^^^^^^^^
2023-03-24T19:25:51.3624365Z     No traceback found
2023-03-24T19:25:51.3624857Z regnet_y_128gf.D0
2023-03-24T19:25:51.3625189Z ^^^^^^^^^^^^^^^^^
2023-03-24T19:25:51.3625442Z     No traceback found
2023-03-24T19:25:51.3625663Z regnet_y_128gf.D1
2023-03-24T19:25:51.3625878Z ^^^^^^^^^^^^^^^^^
2023-03-24T19:25:51.3626070Z     No traceback found
2023-03-24T19:25:51.3626279Z resnet152.0
2023-03-24T19:25:51.3626673Z ^^^^^^^^^^^
2023-03-24T19:25:51.3626895Z     No traceback found
2023-03-24T19:25:51.3627098Z Summary
2023-03-24T19:25:51.3627586Z -------
2023-03-24T19:25:51.3627794Z     Success: 32
2023-03-24T19:25:51.3628007Z     Failures: 5
```

**Comment by Delaunay** [comment]
```
2023-03-27T14:44:35.8526655Z Error Report
2023-03-27T14:44:35.8526886Z ------------
2023-03-27T14:44:35.8526999Z 
2023-03-27T14:44:35.8527176Z resnet152.0
2023-03-27T14:44:35.8527370Z ^^^^^^^^^^^
2023-03-27T14:44:35.8527554Z     No traceback found
2023-03-27T14:44:35.8527726Z Summary
2023-03-27T14:44:35.8527931Z -------
2023-03-27T14:44:35.8528114Z     Success: 23
2023-03-27T14:44:35.8528290Z     Failures: 1
2023-03-27T14:44:35.8528732Z [DONE] Reports directory: /opt/actions-runner/_work/milabench/milabench/output/runs/demepiki.2023-03-27_10:35:18.434278
2023-03-27T14:44:35.8529366Z Source: /opt/actions-runner/_work/milabench/milabench/output/runs/demepiki.2023-03-27_10:35:18.434278
2023-03-27T14:44:35.8529645Z =================
2023-03-27T14:44:35.8529813Z Benchmark results
2023-03-27T14:44:35.8529997Z =================
2023-03-27T14:44:35.8530234Z                 n fail       perf   perf_adj   std%   sem% peak_memory
2023-03-27T14:44:35.8530482Z bert            2    0      30.48      30.48  30.1%   6.4%        9180
2023-03-27T14:44:35.8530776Z convnext_large  2    0      41.91      41.91  19.4%   4.1%        6536
2023-03-27T14:44:35.8531136Z efficientnet_b0 2    0     131.35     131.35  46.4%   9.9%        1299
2023-03-27T14:44:35.8531375Z efficientnet_b4 2    0     129.07     129.07  46.9%  10.0%        4588
2023-03-27T14:44:35.8531628Z efficientnet_b7 2    0      70.26      70.26  36.7%   7.6%        9829
2023-03-27T14:44:35.8531878Z ppo             1    0   23406.39   23406.39  14.6%   4.4%        8096
2023-03-27T14:44:35.8532114Z reformer        2    0       6.35       6.35  47.7%  10.2%        6691
2023-03-27T14:44:35.8532345Z regnet_y_128gf  2    0      18.29      18.29  25.3%   5.4%        6698
2023-03-27T14:44:35.8532653Z resnet152       1    1        NaN        NaN    NaN    NaN          -1
2023-03-27T14:44:35.8533020Z resnet50        2    0     144.29     144.29  34.3%   7.3%        1461
2023-03-27T14:44:35.8533251Z squeezenet1_1   2    0     142.93     142.93  27.6%   5.9%         548
2023-03-27T14:44:35.8533490Z t5              2    0       7.39       7.39  32.2%   6.9%        6548
2023-03-27T14:44:35.8533715Z td3             1    0   14733.76   14733.76  17.0%   5.1%        8096
2023-03-27T14:44:35.8533929Z vit_l_32        1    0      33.63      33.63  12.1%   3.7%        8097
2023-03-27T14:44:35.8534177Z Traceback (most recent call last):
2023-03-27T14:44:35.8534609Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/bin/milabench", line 6, in <module>
2023-03-27T14:44:35.8535209Z     sys.exit(main())
2023-03-27T14:44:35.8535589Z   File "/opt/actions-runner/_work/milabench/milabench/milabench/cli.py", line 41, in main
2023-03-27T14:44:35.8535884Z     sys.exit(run_cli(Main, argv=argv))
2023-03-27T14:44:35.8536361Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/coleo/cli.py", line 628, in run_cli
2023-03-27T14:44:35.8624202Z     return call(opts=opts, args=args)
2023-03-27T14:44:35.8626090Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/coleo/cli.py", line 587, in thunk
2023-03-27T14:44:35.8626481Z     result = fn(*args)
2023-03-27T14:44:35.8627077Z   File "/opt/actions-runner/_work/milabench/milabench/milabench/cli.py", line 326, in run
2023-03-27T14:44:35.8627362Z     make_report(
2023-03-27T14:44:35.8627772Z   File "/opt/actions-runner/_work/milabench/milabench/milabench/report.py", line 226, in make_report
2023-03-27T14:44:35.8729247Z     df = _report_pergpu(summary, measure=measure)
2023-03-27T14:44:35.8730126Z   File "/opt/actions-runner/_work/milabench/milabench/milabench/report.py", line 151, in _report_pergpu
2023-03-27T14:44:35.8730473Z     maxes = df.loc[:, list(range(ngpus))].max(axis=1).transpose()
2023-03-27T14:44:35.8731028Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexing.py", line 1067, in __getitem__
2023-03-27T14:44:35.8780645Z     return self._getitem_tuple(key)
2023-03-27T14:44:35.8781439Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexing.py", line 1256, in _getitem_tuple
2023-03-27T14:44:35.8782037Z     return self._getitem_tuple_same_dim(tup)
2023-03-27T14:44:35.8782610Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexing.py", line 924, in _getitem_tuple_same_dim
2023-03-27T14:44:35.8783159Z     retval = getattr(retval, self.name)._getitem_axis(key, axis=i)
2023-03-27T14:44:35.8783716Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexing.py", line 1301, in _getitem_axis
2023-03-27T14:44:35.8785217Z     return self._getitem_iterable(key, axis=axis)
2023-03-27T14:44:35.8785863Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexing.py", line 1239, in _getitem_iterable
2023-03-27T14:44:35.8786383Z     keyarr, indexer = self._get_listlike_indexer(key, axis)
2023-03-27T14:44:35.8787001Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexing.py", line 1432, in _get_listlike_indexer
2023-03-27T14:44:35.8788490Z     keyarr, indexer = ax._get_indexer_strict(key, axis_name)
2023-03-27T14:44:35.8789434Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 6070, in _get_indexer_strict
2023-03-27T14:44:35.8925334Z     self._raise_if_missing(keyarr, indexer, axis_name)
2023-03-27T14:44:35.8925984Z   File "/opt/actions-runner/_work/milabench/milabench/3/envs/test/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 6130, in _raise_if_missing
2023-03-27T14:44:35.8932159Z     raise KeyError(f"None of [{key}] are in the [{axis_name}]")
2023-03-27T14:44:35.8932665Z KeyError: "None of [Int64Index([0, 1], dtype='int64')] are in the [columns]"
2023-03-27T14:44:36.0892387Z ##[error]Process completed with exit code 1.
```

**Comment by Delaunay** [comment]
```
 squeezenet1_1.D0 [stderr] Traceback (most recent call last):
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/bin/voir", line 8, in <module>
squeezenet1_1.D0 [stderr]     sys.exit(main())
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/voir/cli.py", line 30, in main
squeezenet1_1.D0 [stderr]     ov(sys.argv[1:] if argv is None else argv)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/voir/overseer.py", line 148, in __call__
squeezenet1_1.D0 [stderr]     super().__call__(*args, **kwargs)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/voir/phase.py", line 326, in __call__
squeezenet1_1.D0 [stderr]     super().__call__(*args, **kwargs)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/voir/phase.py", line 268, in __call__
squeezenet1_1.D0 [stderr]     self.run(*args, **kwargs)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/voir/overseer.py", line 132, in run
squeezenet1_1.D0 [stderr]     self.argparser.merge_base_config(yaml.safe_load(open(config, "r")))
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/voir/argparse_ext.py", line 129, in merge_base_config
squeezenet1_1.D0 [stderr]     self.base_configs = OmegaConf.merge(self.base_configs, config)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/omegaconf.py", line 273, in merge
squeezenet1_1.D0 [stderr]     target.merge_with(*configs[1:])
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/basecontainer.py", line 492, in merge_with
squeezenet1_1.D0 [stderr]     self._format_and_raise(key=None, value=None, cause=e)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/base.py", line 231, in _format_and_raise
squeezenet1_1.D0 [stderr]     format_and_raise(
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/_utils.py", line 899, in format_and_raise
squeezenet1_1.D0 [stderr]     _raise(ex, cause)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/_utils.py", line 797, in _raise
squeezenet1_1.D0 [stderr]     raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/basecontainer.py", line 490, in merge_with
squeezenet1_1.D0 [stderr]     self._merge_with(*others)
squeezenet1_1.D0 [stderr]   File "/opt/actions-runner/_work/milabench/milabench/output/venv/torch/lib/python3.9/site-packages/omegaconf/basecontainer.py", line 506, in _merge_with
squeezenet1_1.D0 [stderr]     raise ValueError("Cannot merge with a None config")
squeezenet1_1.D0 [stderr] ValueError: Cannot merge with a None config
squeezenet1_1.D0 [stderr]     full_key:
squeezenet1_1.D0 [stderr]     object_type=dict
squeezenet1_1.D0 ValueError: Cannot merge with a None config
    full_key: 
    object_type=dict
```

**Review by breuleux — APPROVED** [review]

**Pull request merged**